### PR TITLE
[Mime] sync message serializer code for forward-compatibility

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
Before continuing here we should IMO discuss whether or not the BC promise applies to the serialized message format or if #47462 would introduce a BC break.

ping @fabpot

**Comment by fabpot** [comment]
As the removed information was never used anyway, I think it's safe.

**Review by fabpot — APPROVED** [review]

**Comment by xabbuh** [comment]
@fabpot I think I need to look into this a bit more. The `deps=low` job shouldn’t fail. I’ll take care of this later today.

File: src/Symfony/Component/Mime/composer.json [hunk 1]
```diff
@@ -26,13 +26,14 @@
         "symfony/dependency-injection": "^5.4|^6.0",
         "symfony/property-access": "^5.4|^6.0",
         "symfony/property-info": "^5.4|^6.0",
-        "symfony/serializer": "^5.4|^6.0"
+        "symfony/serializer": "^6.1.6"
```
Comment (xabbuh) [inline src/Symfony/Component/Mime/composer.json]: Should we bump dependencies in the `composer.json` or make the changes on 5.4?

Comment (nicolas-grekas) [reply src/Symfony/Component/Mime/composer.json]: let's backport on 5.4, that'd prevent anyone from relying on these keys by mistake

File: src/Symfony/Component/Mime/composer.json [hunk 2]
```diff
@@ -26,13 +26,14 @@
...
     },
     "conflict": {
         "egulias/email-validator": "~3.0.0",
         "phpdocumentor/reflection-docblock": "<3.2.2",
         "phpdocumentor/type-resolver": "<1.4.0",
-        "symfony/mailer": "<5.4"
+        "symfony/mailer": "<5.4",
+        "symfony/serializer": "<6.1.6>"
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Mime/composer.json]: bad range :)

Comment (xabbuh) [reply src/Symfony/Component/Mime/composer.json]: 🙈

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**